modules/drivers/presto-common/src/metabase/driver/presto_common.clj

```diff
@@ -0,0 +1,203 @@
+     [#"(?i)varbinary.*"                :type/*]
+     [#"(?i)json"                       :type/Text] ; TODO - this should probably be Dictionary or something
+     [#"(?i)date"                       :type/Date]
+     [#"(?i)^timestamp$"                :type/DateTime]
+     [#"(?i)^timestamp with time zone$" :type/DateTimeWithTZ]
+     [#"(?i)^time$"                     :type/Time]
+     [#"(?i)^time with time zone$"      :type/TimeWithTZ]
+     #_[#"(?i)time.+"                     :type/DateTime] ; TODO - get rid of this one?
+     [#"(?i)array"                      :type/Array]
+     [#"(?i)map"                        :type/Dictionary]
+     [#"(?i)row.*"                      :type/*] ; TODO - again, but this time we supposedly have a schema
+     [#".*"                             :type/*]]))
+
+(defmethod sql.qp/add-interval-honeysql-form :presto-common
+  [_ hsql-form amount unit]
+  (hsql/call :date_add (hx/literal unit) amount hsql-form))
+
+(defn describe-catalog-sql
+  "The SHOW SCHEMAS statement that will list all schemas for the given `catalog`."
+  {:added "0.39.0"}
+  [driver catalog]
+  (str "SHOW SCHEMAS FROM " (sql.u/quote-name driver :database catalog)))
+
+(defn describe-schema-sql
+  "The SHOW TABLES statement that will list all tables for the given `catalog` and `schema`."
+  {:added "0.39.0"}
+  [driver catalog schema]
+  (str "SHOW TABLES FROM " (sql.u/quote-name driver :schema catalog schema)))
+
+(defn describe-table-sql
+  "The DESCRIBE  statement that will list information about the given `table`, in the given `catalog` and schema`."
+  {:added "0.39.0"}
+  [driver catalog schema table]
+  (str "DESCRIBE " (sql.u/quote-name driver :table catalog schema table)))
+
+(def excluded-schemas
+  "The set of schemas that should be excluded when querying all schemas."
+  #{"information_schema"})
+
+(defmethod driver/db-start-of-week :presto-common
+  [_]
+  :monday)
+
+(defmethod sql.qp/->honeysql [:presto-common Boolean]
+  [_ bool]
+  (hsql/raw (if bool "TRUE" "FALSE")))
+
+(defmethod sql.qp/->honeysql [:presto-common :time]
+  [_ [_ t]]
+  (hx/cast :time (u.date/format-sql (t/local-time t))))
+
+(defmethod sql.qp/->float :presto-common
+  [_ value]
+  (hx/cast :double value))
+
+(defmethod sql.qp/->honeysql [:presto-common :regex-match-first]
+  [driver [_ arg pattern]]
+  (hsql/call :regexp_extract (sql.qp/->honeysql driver arg) (sql.qp/->honeysql driver pattern)))
+
+(defmethod sql.qp/->honeysql [:presto-common :median]
+  [driver [_ arg]]
+  (hsql/call :approx_percentile (sql.qp/->honeysql driver arg) 0.5))
+
+(defmethod sql.qp/->honeysql [:presto-common :percentile]
+  [driver [_ arg p]]
+  (hsql/call :approx_percentile (sql.qp/->honeysql driver arg) (sql.qp/->honeysql driver p)))
+
+;; Presto mod is a function like mod(x, y) rather than an operator like x mod y
+(defmethod hformat/fn-handler (u/qualified-name ::mod)
+  [_ x y]
+  (format "mod(%s, %s)" (hformat/to-sql x) (hformat/to-sql y)))
```

Comment:
can you not do `(hsql/call :mod x y)`?

Reply:
Apparently not.  The `hsql/call` ends up calling **this** (`fn-handler`) to figure out the "form" the function invocation could take.  There are a couple other drivers that do this exact same thing (I copied the method from there; see Oracle and BigQuery).  To be honest, though, I'm unsure where the `x mod y` (i.e. the default) form is even defined.  I tried to find it and couldn't.